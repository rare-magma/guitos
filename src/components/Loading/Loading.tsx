import { Container, Row, Spinner } from "react-bootstrap";

function Loading() {
  return (
    <Container
      fluid
      className="position-absolute top-50 start-50 translate-middle"
    >
      <Row className="justify-content-center">
        <Spinner animation="border" role="status" />
      </Row>
    </Container>
  );
}

export default Loading;
