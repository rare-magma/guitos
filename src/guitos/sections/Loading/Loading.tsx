import { Container, Row, Spinner } from "react-bootstrap";

export function Loading() {
  return (
    <Container
      fluid={true}
      className="position-absolute top-50 start-50 translate-middle"
    >
      <Row className="justify-content-center">
        {/** biome-ignore lint/a11y/useSemanticElements: bootstrap isn't semantic */}
        <Spinner animation="border" role="status" />
      </Row>
    </Container>
  );
}
