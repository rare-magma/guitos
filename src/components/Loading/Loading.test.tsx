import { render, screen } from "@testing-library/react";
import Loading from "./Loading";

describe("Loading", () => {
  const comp = <Loading />;

  beforeEach(() => {
    render(comp);
  });

  it("matches snapshot", () => {
    expect(comp).toMatchSnapshot();
  });

  it("renders initial state", () => {
    expect(screen.getByRole("status")).toBeInTheDocument();
  });
});
