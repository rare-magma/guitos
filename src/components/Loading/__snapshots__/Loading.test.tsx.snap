// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading > matches snapshot 1`] = `<Loading />`;
